src/rgw/rgw_log.cc

```diff
@@ -226,6 +225,18 @@ static void log_usage(struct req_state *s, const string& op_name)
   utime_t ts = ceph_clock_now();
 
   usage_logger->insert(ts, entry);
+
+  if(s->cct->_conf->rgw_enable_usage_log_at_subuser_level) {
+    const auto subuser = s->auth.identity->get_subuser_name();
+
+    if (subuser) {
+      std::string sname = u + ":" + *subuser;
+      ldout(s->cct, 5) << "log subuser entry: " << sname << dendl;
+      rgw_usage_log_entry sentry(sname, p, bucket_name);
```

Comment:
Sounds reasonable. In order to resolve the conflict between`<tenant>$<user>` and `<user>:<subuser>` and backward compatible , we record subuser related statistics with a prefix `$`  in omap.

subuser's [log_add_by_time](
https://github.com/ceph/ceph/pull/13929/files#diff-c220b6b87cf8c4612b3582349b124112R2748) ，[log_add_by_user](https://github.com/ceph/ceph/pull/13929/files#diff-c220b6b87cf8c4612b3582349b124112R2748) would record separately with user's.

The prefix `$` is selected cause by [$ is the separator between tenant and user](https://github.com/ceph/ceph/blob/master/src/rgw/rgw_basic_types.h#L37).If the keystone user create a tenant that name start with a `$`  , we may still has conflict issue. The root cause of such conflict is the usage record in omap used user-inputed `rgw_user.to_str()` . Maybe we shouldn't trust user input, use uuid instead? 

Looking forward for you input. Thx.